app/models/contact.rb

```diff
@@ -63,14 +63,20 @@ def generate_request
   def receive_shareable(shareable)
     ShareVisibility.create!(:shareable_id => shareable.id, :shareable_type => shareable.class.base_class.to_s, :contact_id => self.id)
   end
+  
+  def incoming_aspects
+    Aspect.where(
+          :user_id => self.person.owner_id,
+          :contacts_visible => true).joins(:contacts).where(
+            :contacts => {:person_id => self.user.person_id}).select('aspects.id')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.